Upstream-Status: Pending

From 09816ba9f6351881af8015fc2703707eba27d5c8 Mon Sep 17 00:00:00 2001
From: Rob Woolley <rob.woolley@windriver.com>
Date: Sun, 2 Mar 2025 09:30:52 -0800
Subject: [PATCH] Simplify removing leading and trailing separators

Previously, we checked if the leading or trailing character was a colon
and then used a wildcard to remove it.

It is simpler to just remove a leading or trailing colon. This has the
added benefit of only using shell built-in functions.

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>
---
 ament_package/template/prefix_level/_local_setup_util.py | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/ament_package/template/prefix_level/_local_setup_util.py b/ament_package/template/prefix_level/_local_setup_util.py
index 813bd8a..bcb3040 100644
--- a/ament_package/template/prefix_level/_local_setup_util.py
+++ b/ament_package/template/prefix_level/_local_setup_util.py
@@ -48,10 +48,8 @@ def main(argv=sys.argv[1:]):  # noqa: D103
         FORMAT_STR_USE_ENV_VAR = '${name}'
         FORMAT_STR_INVOKE_SCRIPT = 'AMENT_CURRENT_PREFIX="{prefix}" ' \
             '_ament_prefix_sh_source_script "{script_path}"'
-        FORMAT_STR_REMOVE_LEADING_SEPARATOR = 'if [ "$(echo -n ${name} | ' \
-            'head -c 1)" = ":" ]; then export {name}=${{{name}#?}} ; fi'
-        FORMAT_STR_REMOVE_TRAILING_SEPARATOR = 'if [ "$(echo -n ${name} | ' \
-            'tail -c 1)" = ":" ]; then export {name}=${{{name}%?}} ; fi'
+        FORMAT_STR_REMOVE_LEADING_SEPARATOR = 'export {name}=${{{name}#:}}'
+        FORMAT_STR_REMOVE_TRAILING_SEPARATOR = 'export {name}=${{{name}%:}}'
     elif args.primary_extension == 'bat':
         FORMAT_STR_COMMENT_LINE = ':: {comment}'
         FORMAT_STR_SET_ENV_VAR = 'set "{name}={value}"'
-- 
2.43.0

